src/views/shared/client/memory_data_loader.test.ts

```diff
@@ -0,0 +1,34 @@
+import { MemoryDataLoader } from "./memory_data_loader";
+
+test("Tests injectMapData, getMapData, and getAnyMapData with two different map types.", () => {
+  const md = new MemoryDataLoader();
+  md.injectMapData("IDToName", {
+    "1": "T1",
+    "2": "T2",
+  });
+  md.injectMapData("IDToToughness", {
+    1: "One",
+    2: "Two",
+    3: "Three",
+  });
+  expect(md.getMapData("IDToName")).toStrictEqual({ "1": "T1", "2": "T2" });
+  expect(md.getAnyMapData("IDToName")).toStrictEqual({ "1": "T1", "2": "T2" });
+  expect(md.getMapData("IDToToughness")).toStrictEqual({
+    1: "One",
+    2: "Two",
+    3: "Three",
+  });
+  expect(md.getAnyMapData("IDToToughness")).toStrictEqual({
+    1: "One",
+    2: "Two",
+    3: "Three",
+  });
+});
+
+test("Ensures that onLoaded, onLoadedAll, and isDoneLoading return immediately", () => {
```

Comment:
All functions in javacsript "return immediately" when you call them. It'd be better to test that the callbacks are ran immediately:

```ts
let callbackCallCount = 0;
md.onLoadedCB("T1", () =>{ callbackCallCount++ });
md.onAllLoadedCB(["T1"], () =>{ callbackCallCount++ });
expect(callbackCallCount).toBe(2);
```

The other thing in here, `expect(md.isDoneLoading("T1" as any)).toBe(true);` makes me think you meant to name this test "Always reports true when asked if a particular map is loaded." Buuutttt I don't think that's a great thing, it should probably return true if the data is injected. And for the callback ones, they should probably only the callback function until the data is injected, and then call the callbacks.

If you don't want to implement that, you could just put it in a TODO in the memory_data_loader.ts file.